frontstage/views/secure_messaging.py

```diff
@@ -69,15 +69,17 @@ def create_message(session):
         if request.form['submit'] == 'Save draft':
             if "msg_id" in request.form:
                 data['msg_id'] = request.form['msg_id']
+                headers['Authorization'] = request.cookies['authorization']
                 response = requests.put(DRAFT_PUT_API_URL.format(request.form['msg_id']), data=json.dumps(data), headers=headers)
                 if response.status_code != 200:
-                    # TODO replace with custom error page when available
-                    return redirect(url_for('error_bp.error_page'))
+                    get_json = json.loads(response.content)
+                    return render_template('secure-messages-draft.html', _theme='default', draft=data, errors=get_json)
+
             else:
                 response = requests.post(DRAFT_SAVE_API_URL, data=json.dumps(data), headers=headers)
                 if response.status_code != 201:
```

Comment:
This probably needs to be more specific, If the api returns a 404, 500 we dont want to be returning the error messages